### PR TITLE
fix: webui trial chart render metrics with same name [DET-4169]

### PR DESCRIPTION
**Pull request by apizzini**
## Description

Metrics with the same name were conflicting in charts.
Issue fixed by grouping metrics with "<type>_<name>".

Schermata 2020-09-25 alle 17 22 57

Schermata 2020-09-25 alle 17 23 01

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

### CODE REVIEW

File: webui/react/src/pages/TrialDetails/TrialChart.tsx [hunk 1]
```diff
@@ -44,26 +44,28 @@ const TrialChart: React.FC<Props> = ({
         const y = metric.type === MetricType.Validation ?
           validationSource[metric.name] : trainingSource[metric.name];
 
+        const metricKey = metric.type+'_'+metric.name;
```
Comment (hkang1) [inline webui/react/src/pages/TrialDetails/TrialChart.tsx:47]: **nit:** Super minor thing, but we try to use template literals whenever possible.
```
const metricKey = `${metric.type}_${metric.name}`;
```

File: webui/react/src/pages/TrialDetails/TrialChart.tsx [hunk 2]
```diff
@@ -44,26 +44,28 @@ const TrialChart: React.FC<Props> = ({
...
 
         if (text && x && y) {
-          if (!dataMap[metric.name]) {
-            dataMap[metric.name] = {
+          if (!dataMap[metricKey]) {
+            dataMap[metricKey] = {
               hovertemplate: '%{text}<extra></extra>',
               mode: 'lines+markers',
-              name: metric.name,
+              name: metric.name+' ('+metric.type+')',
```
Comment (hkang1) [inline webui/react/src/pages/TrialDetails/TrialChart.tsx:59]: **nit:** ditto
```
name: `${metric.name} (${metric.type})`,
```

**Review by hkang1 — APPROVED** [review]
LGTM! Sweet fix! couple of minor nit picks but work great. This is a user facing fix, we'll need to have release notes for this. To add one, add a file under the directory `/docs/release-notes`. Basically you'll add a text file named as `<PR #>-description-text.txt`. So for this specific one will be something like `1350-render-metrics-chart-with-same-same.txt`. Take a look at the `README.md` in that directory on specific guidance on what to write, but it will look something like this:
```
:orphan:

**BUG FIXES**

- WebUI: Fix rendering metrics with the same name on the metric chart.
```

File: webui/react/src/pages/TrialDetails/TrialChart.tsx (hunk 2 above)
Comment (apizzini) [reply webui/react/src/pages/TrialDetails/TrialChart.tsx]: Thanks for the review. Changed.

File: webui/react/src/pages/TrialDetails/TrialChart.tsx (hunk 1 above)
Comment (apizzini) [reply webui/react/src/pages/TrialDetails/TrialChart.tsx]: Thanks for the review. Changed.

**Pull request merged**